code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -464,3 +434,335 @@
+
+/datum/reagent/drug/saturnx/on_mob_metabolize(mob/living/L)
+	. = ..()
+	playsound(L, 'sound/chemistry/saturnx_fade.ogg', 30)
+	to_chat(L, span_nicegreen("You feel pins and needles all over your skin as your body suddenly becomes transparent!"))
+	addtimer(CALLBACK(src, .proc/turn_man_invisible, L), 10) //just a quick delay to synch up the sound.
+	if(!L.hud_used)
+		return
+
+	var/atom/movable/plane_master_controller/game_plane_master_controller = L.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+
+	var/list/col_filter_full = list(1,0,0,0, 0,1.00,0,0, 0,0,1,0, 0,0,0,1, 0,0,0,0)
+	var/list/col_filter_twothird = list(1,0,0,0, 0,0.68,0,0, 0,0,1,0, 0,0,0,1, 0,0,0,0)
+	var/list/col_filter_half = list(1,0,0,0, 0,0.42,0,0, 0,0,1,0, 0,0,0,1, 0,0,0,0)
+	var/list/col_filter_empty = list(1,0,0,0, 0,0,0,0, 0,0,1,0, 0,0,0,1, 0,0,0,0)
+
+	game_plane_master_controller.add_filter("saturnx_filter", 10, color_matrix_filter(col_filter_twothird, FILTER_COLOR_HCY))
+
+	for(var/filter in game_plane_master_controller.get_filters("saturnx_filter"))
+		animate(filter, loop = -1, color = col_filter_full, time = 4 SECONDS, easing = CIRCULAR_EASING|EASE_IN, flags = ANIMATION_PARALLEL)
+		//uneven so we spend slightly less time with bright colors
+		animate(color = col_filter_twothird, time = 6 SECONDS, easing = LINEAR_EASING)
+		animate(color = col_filter_half, time = 3 SECONDS, easing = LINEAR_EASING)
+		animate(color = col_filter_empty, time = 2 SECONDS, easing = CIRCULAR_EASING|EASE_OUT)
+		animate(color = col_filter_half, time = 24 SECONDS, easing = CIRCULAR_EASING|EASE_IN)
+		animate(color = col_filter_twothird, time = 12 SECONDS, easing = LINEAR_EASING)
+
+	game_plane_master_controller.add_filter("saturnx_blur", 1, list("type" = "radial_blur", "size" = 0))
+
+	for(var/filter in game_plane_master_controller.get_filters("saturnx_blur"))
+		animate(filter, loop = -1, size = 0.04, time = 2 SECONDS, easing = ELASTIC_EASING|EASE_OUT, flags = ANIMATION_PARALLEL)
+		animate(size = 0, time = 6 SECONDS, easing = CIRCULAR_EASING|EASE_IN)
+
+///This proc turns the guy who took the drug invisible by giving him the invisible man trait and updating his body, this changes the sprite of all his organic limbs to a 1 alpha version.
+/datum/reagent/drug/saturnx/proc/turn_man_invisible(mob/living/carbon/invisible_man)
+	if(!invisible_man.getorganslot(ORGAN_SLOT_LIVER))
+		return
+	if(invisible_man.undergoing_liver_failure())
+		return
+	if(HAS_TRAIT(invisible_man, TRAIT_NOMETABOLISM))
+		return
+
+	ADD_TRAIT(invisible_man, TRAIT_INVISIBLE_MAN, name)
+
+	var/datum/dna/druggy_dna = invisible_man.has_dna()
+	if(druggy_dna?.species)
+		druggy_dna.species.species_traits += NOBLOODOVERLAY
+
+	invisible_man.update_body()
+	invisible_man.update_hair()
+	invisible_man.remove_from_all_data_huds()
+	invisible_man.sound_environment_override = SOUND_ENVIROMENT_PHASED
+
+/datum/reagent/drug/saturnx/on_mob_end_metabolize(mob/living/M)
```

Comment:
please name `M` better

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names